### PR TITLE
Adjust closure initialization for primary constructor

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@cston, @jjonescz, @dotnet/roslyn-compiler Please review

**Comment by AlekseyTs** [comment]
@cston, @jjonescz, @dotnet/roslyn-compiler Please review

File: src/Compilers/CSharp/Test/Semantic/Semantics/PrimaryConstructorTests.cs [hunk 1]
```diff
@@ -10261,7 +10245,7 @@ struct S1(Color Color)
 {
     public S1() : this(
         Color.M1(
-            new Color()))
```
Comment (jjonescz) [inline src/Compilers/CSharp/Test/Semantic/Semantics/PrimaryConstructorTests.cs:10264]: >new Color() [](http://example.com/codeflow?start=12&length=11)

Should we test that the ambiguity error is raised even when one method is inapplicable due to argument type mismatch, e.g., calling `Color.M1(new Color())` in this test?

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Semantic/Semantics/PrimaryConstructorTests.cs:10264]: > Should we test that the ambiguity error is raised even when one method is inapplicable due to argument type mismatch, e.g., calling Color.M1(new Color()) in this test?

I think inapplicability scenarios are adequately covered by other targeted tests. For example, ```ParameterCapturing_027_ColorColor_MemberAccess_InstanceInapplicableAndStaticApplicableDueToArguments_Method```. Also look for other tests with "Inapplicable" in the name. There is also ```ParameterCapturing_063_ColorColor_MemberAccess_InstanceAndStaticAmbiguity_Method_ApplicabilityDueToArgumentsNotConsidered```.

**Comment by AlekseyTs** [comment]
@cston, @dotnet/roslyn-compiler For the second review.

**Review by cston — APPROVED** [review]

**Pull request merged**